cmd/notation/registry.go

```diff
@@ -127,3 +179,36 @@ func getSavedCreds(ctx context.Context, serverAddress string) (auth.Credential,
 
 	return nativeStore.Get(serverAddress)
 }
+
+func pingReferrersAPI(ctx context.Context, remoteRepo *remote.Repository) error {
+	if err := remoteRepo.SetReferrersCapability(true); err != nil {
+		return err
+	}
+	var checkReferrerDesc ocispec.Descriptor
+	checkReferrerDesc.Digest = zeroDigest
+	// core process
+	err := remoteRepo.Referrers(ctx, checkReferrerDesc, "", func(referrers []ocispec.Descriptor) error {
+		return nil
+	})
+	if err != nil {
+		var errResp *errcode.ErrorResponse
+		if !errors.As(err, &errResp) || errResp.StatusCode != http.StatusNotFound {
+			return err
+		}
+		if isErrorCode(errResp, errcode.ErrorCodeNameUnknown) {
+			// The repository is not found.
+			return err
+		}
+		// A 404 returned by Referrers API indicates that Referrers API is
+		// not supported.
+		errMsg := fmt.Sprintf("failed to ping Referrers API with error: %v. This is due to target registry does not support the Referrers API. Try store signatures with OCI image manifest using `--signature-manifest image`", err)
```

Comment:
The error message can be simplified as
```suggestion
		errMsg := "Target registry does not support the Referrers API. Try the flag `--signature-manifest image` for backwards compatibility", err)
```
where the detailed error message can be in the log.

Reply:
To be consistent should we use `Try the flag --signature-manifest image to store signatures using OCI image manifest` here ?

Reply:
I think we could combine both, i.e. `Try the flag --signature-manifest image to store signatures using OCI image manifest for backwards compatibility`. The benefit to include `for backwards compatibility` is that a user might not know what `OCI image manifest` is. Adding `for backwards compatibility` shows the purpose behind this flag.